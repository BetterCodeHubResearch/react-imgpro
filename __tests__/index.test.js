import React from 'react';
import renderer from 'react-test-renderer';
import getSize from '../src/utils/getDimensions';

describe('getDimension', () => {
  const inp = { resize: { height: 300, width: 400 }};
  const def = 300;

  test('should return the image height when attribute passed is \'height\' ', () => {  
    const height = getSize(inp, def, 'height');
  
    expect(height).toMatchSnapshot();
  })

  test('should return the image width when attribute passed is \'width\' ', () => {  
    const width = getSize(inp, def, 'width');
  
    expect(width).toMatchSnapshot();
  });

  test('should fallback to default', () => {  
    const height = getSize({}, def, 'height');
  
    expect(height).toMatchSnapshot();
  });
});
