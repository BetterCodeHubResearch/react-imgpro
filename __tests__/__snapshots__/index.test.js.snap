// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDimension should fallback to default 1`] = `300`;

exports[`getDimension should return the image height when attribute passed is 'height'  1`] = `300`;

exports[`getDimension should return the image width when attribute passed is 'width'  1`] = `400`;
